samza-kafka/src/main/scala/org/apache/samza/config/KafkaConfig.scala

```diff
@@ -320,14 +316,17 @@ class KafkaProducerConfig(val systemName: String,
       producerProperties.put(ProducerConfig.MAX_IN_FLIGHT_REQUESTS_PER_CONNECTION, MAX_IN_FLIGHT_REQUESTS_PER_CONNECTION_DEFAULT)
     }
 
-    if (producerProperties.containsKey(ProducerConfig.RETRIES_CONFIG)
-      && producerProperties.get(ProducerConfig.RETRIES_CONFIG).asInstanceOf[String].toInt < RETRIES_DEFAULT) {
-      warn("Samza does not provide producer failure handling. Consider setting '%s' to a large value, like Int.MAX." format ProducerConfig.RETRIES_CONFIG)
-    } else {
-      // Retries config is set to Max so that when all attempts fail, Samza also fails the send. We do not have any special handler
-      // for producer failure
+    if (!producerProperties.containsKey(ProducerConfig.RETRIES_CONFIG)) {
```

Comment:
Checked with the Kafka team. Your understanding is correct, it will ultimately fail with expired batches after a couple mins. 
They said the behavior will be roughly equivalent, since the batch will likely expire before 20 retries or INT_MAX retries. 
To err on the side of caution, I'll change it back to the previous value